src/storage/src/source/mysql/snapshot.rs

```diff
@@ -517,6 +517,22 @@ where
     Ok(total)
 }
 
+/// Builds the SQL query to be used for creating the snapshot using the first entry in outputs.
+///
+/// Expect `outputs` to contain entries for a single table, and to have at least 1 entry.
+/// Expect that each MySqlTableDesc entry contains all columns described in information_schema.columns.
+#[must_use]
+fn build_snapshot_query(outputs: &[SourceOutputInfo]) -> String {
+    let info = outputs.first().expect("MySQL table info");
+    let columns = info
+        .desc
+        .columns
+        .iter()
+        .map(|col| format!("`{}`", col.name))
+        .join(", ");
+    format!("SELECT {} FROM {}", columns, info.table_name)
```

Comment:
(sorry for the drive-by review) Do we have to worry about SQL injection here?

Reply:
I did wonder about that during review and concluded that we don't have to be worried as to perform an injection you have to somehow create weirdly named columns, which means you already admin level have access to the database. My understanding is that we can't use prepared statements here since with those you can put placeholders for values but here we want to put placeholders for column names. If it's possible and easy to do we should maybe do it

Reply:
nice!